src/main/java/com/philemonworks/selfdiagnose/output/HTTPReporter.java

```diff
@@ -0,0 +1,49 @@
+/*
+    Copyright 2008-2011 Ernest Micklei @ PhilemonWorks.com
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+package com.philemonworks.selfdiagnose.output;
+
+import com.philemonworks.selfdiagnose.DiagnosticTaskResult;
+import com.philemonworks.selfdiagnose.SelfDiagnose;
+import com.philemonworks.selfdiagnose.Severity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+
+/**
+ * HTMLReporter is to produce an HTTP report for a DiagnoseRun.
+ *
+ */
+public class HTTPReporter implements DiagnoseRunReporter {
```

Comment:
Better call this StatusReporter as it does not actually report with a HTTP response.

Reply:
Changed.